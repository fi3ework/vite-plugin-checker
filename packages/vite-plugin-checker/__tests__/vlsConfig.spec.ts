import path from 'node:path'
import { describe, expect, it } from 'vitest'
import { ShutdownRequest } from 'vscode-languageserver/node'
import { URI } from 'vscode-uri'

import {
  logLevel2Severity,
  prepareClientConnection,
} from '../src/checkers/vls/diagnostics'

async function testVslConfig(overrideConfig?: any) {
  const workspaceUri = URI.file(path.join(__dirname, 'fixtures'))
  const { clientConnection, serverConnection, vls, up, down } =
    await prepareClientConnection(workspaceUri, logLevel2Severity.WARN, {
      watch: false,
      verbose: false,
      config: overrideConfig || null,
    })

  // @ts-expect-error
  expect(vls.workspaceConfig).toMatchSnapshot()

  // TODO: this test case will let Jest hang with out --forceExit option
  // current enable forceExit but we should find which resource is not released
  await clientConnection.sendRequest(ShutdownRequest.type)
  clientConnection.dispose()
  clientConnection.end()
  serverConnection.dispose()
  up.destroy()
  down.destroy()
  vls.dispose()
}

describe('VLS config', () => {
  it('default config', async () => {
    await testVslConfig()
  })

  it('customized config', async () => {
    await testVslConfig({
      vetur: {
        validation: {
          template: false,
          templateProps: false,
          interpolation: false,
          style: false,
        },
      },
    })
  })
})
