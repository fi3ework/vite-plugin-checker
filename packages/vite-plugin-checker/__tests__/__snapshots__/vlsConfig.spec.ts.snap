// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VLS config > customized config 1`] = `
{
  "css": {},
  "emmet": {},
  "html": {
    "suggest": {},
  },
  "javascript": {
    "format": {},
  },
  "languageStylus": {},
  "stylusSupremacy": {},
  "typescript": {
    "format": {},
    "tsdk": null,
  },
  "vetur": {
    "completion": {
      "autoImport": false,
      "scaffoldSnippetSources": {
        "user": "üóíÔ∏è",
        "vetur": "‚úå",
        "workspace": "üíº",
      },
      "tagCasing": "initial",
    },
    "dev": {
      "logLevel": "INFO",
      "vlsPath": "",
      "vlsPort": -1,
    },
    "experimental": {
      "templateInterpolationService": true,
    },
    "format": {
      "defaultFormatter": {},
      "defaultFormatterOptions": {},
      "enable": true,
      "options": {
        "tabSize": 2,
        "useTabs": false,
      },
      "scriptInitialIndent": false,
      "styleInitialIndent": false,
    },
    "grammar": {
      "customBlocks": {},
    },
    "ignoreProjectWarning": false,
    "languageFeatures": {
      "codeActions": true,
      "semanticTokens": true,
      "updateImportOnFileMove": true,
    },
    "trace": {
      "server": "off",
    },
    "useWorkspaceDependencies": false,
    "validation": {
      "interpolation": false,
      "script": true,
      "style": false,
      "template": false,
      "templateProps": false,
    },
  },
}
`;

exports[`VLS config > default config 1`] = `
{
  "css": {},
  "emmet": {},
  "html": {
    "suggest": {},
  },
  "javascript": {
    "format": {},
  },
  "languageStylus": {},
  "stylusSupremacy": {},
  "typescript": {
    "format": {},
    "tsdk": null,
  },
  "vetur": {
    "completion": {
      "autoImport": false,
      "scaffoldSnippetSources": {
        "user": "üóíÔ∏è",
        "vetur": "‚úå",
        "workspace": "üíº",
      },
      "tagCasing": "initial",
    },
    "dev": {
      "logLevel": "INFO",
      "vlsPath": "",
      "vlsPort": -1,
    },
    "experimental": {
      "templateInterpolationService": true,
    },
    "format": {
      "defaultFormatter": {},
      "defaultFormatterOptions": {},
      "enable": true,
      "options": {
        "tabSize": 2,
        "useTabs": false,
      },
      "scriptInitialIndent": false,
      "styleInitialIndent": false,
    },
    "grammar": {
      "customBlocks": {},
    },
    "ignoreProjectWarning": false,
    "languageFeatures": {
      "codeActions": true,
      "semanticTokens": true,
      "updateImportOnFileMove": true,
    },
    "trace": {
      "server": "off",
    },
    "useWorkspaceDependencies": false,
    "validation": {
      "interpolation": true,
      "script": true,
      "style": true,
      "template": true,
      "templateProps": true,
    },
  },
}
`;
